src/core_plugins/kibana/public/visualize/listing/visualize_listing.js

```diff
@@ -29,7 +29,10 @@ export function VisualizeListingController($injector) {
         this.totalItems = result.total;
         this.showLimitError = result.total > config.get('savedObjects:listingLimit');
         this.listingLimit = config.get('savedObjects:listingLimit');
-        return result.hits;
+        return result.hits.filter(result => {
+          const labMode = config.get('visualize:labMode');
```

Comment:
Move the `config.get` outside of the filter function so the option won't be loaded again for every result.